benchmark/benchmark17_test.go

```diff
@@ -0,0 +1,57 @@
+// +build go1.7
+
+package benchmark
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/benchmark/stats"
+)
+
+func BenchClient(b *testing.B, maxConcurrentCalls, reqSize, respSize int, runMode, enableTrace string) {
+	if strings.EqualFold(enableTrace, "Tracing") {
+		grpc.EnableTracing = true
+	} else {
+		grpc.EnableTracing = false
+	}
+	if strings.EqualFold(runMode, "Unary") {
+		runUnary(b, maxConcurrentCalls, reqSize, respSize)
+	} else {
+		runStream(b, maxConcurrentCalls, reqSize, respSize)
+	}
+}
+
+func BenchmarkClient(b *testing.B) {
+	modeTrace := []struct {
+		runMode     string
+		enableTrace string
+	}{
+		{"Unary", "Tracing"},
+		{"Unary", "noTrace"},
+		{"Stream", "Tracing"},
+		{"Stream", "noTrace"},
+	}
+	maxConcurrentCalls := []int{1, 8, 64, 512}
+	reqSize := []int{1, 1024}
+	respSize := []int{1, 1024}
+	for _, mt := range modeTrace {
```

Comment:
It seems arbitrary to group together the tracing and mode settings, but not the other things.

I'd prefer either:

1. Keep them separate:

```go
for _, mode := range []string{runModeUnary, runModeStreaming} {
  for _, enableTracing := range []bool{false, true} {
    ...
```

2. Or, put everything together in a single testCase struct.  Then this pattern instead:

```go
func testCases() []testCase {
  for _, mode := range []string{runModeUnary, runModeStreaming} {
    for _, enableTracing := range []bool{false, true} {
      ...
              tcs = append(tcs, testCase{mode: mode, enableTracing: enableTracing, ....})

....

func BenchmarkClient() {
  for _, tc := range testCases() {
    b.Run(...)
  }
}
```